extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -554,59 +568,52 @@ class LifecycleTask extends BackbeatTask {
                 (applicableRules, next) => {
                     // if version is latest, only expiration action applies
                     if (version.IsLatest) {
-                        return this._compareObject(
-                            bucketData, version, applicableRules, log, next);
-                    }
-                    const req = this.s3target.headObject({
-                        Bucket: bucketData.target.bucket,
-                        Key: version.Key,
-                        VersionId: version.VersionId,
-                    });
-                    attachReqUids(req, log);
-                    return req.send((err, data) => {
-                        if (err) {
-                            log.error('failed to get object', {
-                                method: 'LifecycleTask._compareVersions',
-                                error: err,
-                                bucket: bucketData.target.bucket,
-                                objectKey: version.Key,
-                                versionId: version.VersionId,
-                            });
-                            return next(err);
-                        }
-                        const objDate = new Date(data.LastModified);
-                        const now = Date.now();
-                        // time diff in ms
-                        const diff = now - objDate;
-                        const daysSinceInitiated =
-                            Math.floor(diff / (1000 * 60 * 60 * 24));
-                        const ncve = 'NoncurrentVersionExpiration';
-                        const ncd = 'NoncurrentDays';
-                        if (applicableRules[ncve] &&
-                        applicableRules[ncve][ncd] &&
-                        daysSinceInitiated >= applicableRules[ncve][ncd]) {
-                            const entry = {
-                                action: 'deleteObject',
-                                target: {
-                                    owner: bucketData.target.owner,
-                                    bucket: bucketData.target.bucket,
-                                    key: version.Key,
-                                    version: data.VersionId,
-                                },
-                            };
-                            this.sendObjectEntry(entry, err => {
-                                if (!err) {
-                                    log.debug('sent object entry for ' +
-                                    'consumption', {
-                                        method: 'LifecycleTask.' +
-                                            '_compareVersions',
-                                        entry,
-                                    });
-                                }
-                            });
+                        // `_compareObject` should not apply for delete markers
+                        if (version.StorageClass && version.Size &&
```

Comment:
Oops, this check is no longer needed as we are ignoring delete markers.

But, it is important to note that ignoring delete markers means staleDate is now compared between only versions and delete markers are not considered. I think it's important to consider them because delete markers are versions themselves.